cmake/flags.cmake

```diff
@@ -132,6 +132,7 @@ set(COMMON_FLAGS
     -Wno-error=int-in-bool-context # Warning in Eigen gcc 7.2
     -Wimplicit-fallthrough=0 # Warning in tinyformat.h
     -Wno-error=maybe-uninitialized # Warning in boost gcc 7.2
+    -Wno-error=parentheses # Warning in boost gcc 8.2
```

Comment:
How about move line134 and line135 to `boost.cmake`? You can finish it in next PR.